src/composables/trade/useRelayerApproval.ts

```diff
@@ -67,6 +80,7 @@ export default function useRelayerApproval(relayerAddress: Ref<string>) {
         onTxConfirmed: () => {
           approving.value = false;
           approved.value = true;
+          relayerApproval.refetch.value();
```

Comment:
it's important to refetch the query when the approval is successful because this query is only fetched once.